binary-install/tests/all/utils/mod.rs

```diff
@@ -0,0 +1,75 @@
+use flate2::write::GzEncoder;
+use flate2::Compression;
+use std::fs::{File, OpenOptions};
+use std::io::{self, Read, Write};
+use std::net::TcpListener;
+use std::sync::mpsc::{channel, Receiver};
+use std::thread;
+
+pub const TEST_SERVER_HOST: &'static str = "localhost";
+
+pub fn start_server(port: u32, tarball: Option<Vec<u8>>) -> Receiver<()> {
+    let (sender, receiver) = channel();
+
+    thread::spawn(move || {
+        TcpListener::bind(format!("{}:{}", TEST_SERVER_HOST, port))
```

Comment:
If you pass 0 as the port, then the OS will assign a random unused one. You can then send the actual port back through the channel. This can be more robust than hard-coding a different value for each test!

See the third paragraph of https://doc.rust-lang.org/std/net/struct.TcpListener.html#method.bind

Reply:
Oh, thanks a lot! Didn't know this, will definitely do it!